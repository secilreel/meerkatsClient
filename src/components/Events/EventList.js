import React from 'react';
import Event from './Event';

export default function EventList (){
    return (
    <div className="events">
    <section className="add-event">
        <button type="button" className="js-add-event-button">Add Event</button>
    </section>
    <div className="events-list">
        <Event/>
    </div>
</div>
);
    }