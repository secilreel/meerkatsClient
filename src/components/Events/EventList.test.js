import React from 'react';
import ReactDOM from 'react-dom';
import EventList from './Event';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {MemoryRouter} from 'react-router-dom';

configure({ adapter: new Adapter() });

describe('<Event />', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <EventList />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
    <MemoryRouter>
        <EventList />
    </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });

})
