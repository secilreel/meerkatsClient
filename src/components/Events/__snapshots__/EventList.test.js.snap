// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> renders the UI as expected 1`] = `
<a
  href="/events/undefined"
  onClick={[Function]}
>
  <section
    className="event container"
  >
    <h2 />
    <div
      className="event-box"
    >
      <img
        alt="headshot of the event planner"
        className="event-owner logo"
      />
      <div
        className="event-summary"
      >
        <div
          className="event-icon-box"
        >
          <img
            alt="calendar icon"
            className="event-icons"
            src="calendar.svg"
          />
          <div>
            Invalid Date
          </div>
        </div>
        <div
          className="event-icon-box"
        >
          <img
            alt="clock icon"
            className="event-icons"
            src="clock.svg"
          />
          <div>
            undefined
          </div>
        </div>
        <div
          className="event-icon-box"
        >
          <img
            alt="map icon"
            className="event-icons"
            src="map.svg"
          />
          <div>
            undefined
          </div>
        </div>
      </div>
    </div>
    <div
      className="button-box"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Remove Event
      </button>
    </div>
  </section>
</a>
`;

exports[`<EventList /> renders the UI as expected 1`] = `
<a
  href="/events/undefined"
  onClick={[Function]}
>
  <section
    className="event container"
  >
    <h2 />
    <div
      className="event-box"
    >
      <img
        alt="headshot of the event planner"
        className="event-owner logo"
      />
      <div
        className="event-summary"
      >
        <div
          className="event-icon-box"
        >
          <img
            alt="calendar icon"
            className="event-icons"
            src="calendar.svg"
          />
          <div>
            Invalid Date
          </div>
        </div>
        <div
          className="event-icon-box"
        >
          <img
            alt="clock icon"
            className="event-icons"
            src="clock.svg"
          />
          <div>
            undefined
          </div>
        </div>
        <div
          className="event-icon-box"
        >
          <img
            alt="map icon"
            className="event-icons"
            src="map.svg"
          />
          <div>
            undefined
          </div>
        </div>
      </div>
    </div>
    <div
      className="button-box"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Remove Event
      </button>
    </div>
  </section>
</a>
`;
